lib/jekyll/document.rb

```diff
@@ -237,18 +218,17 @@ def destination(base_directory)
     def write(dest)
       path = destination(dest)
       FileUtils.mkdir_p(File.dirname(path))
-      File.open(path, 'wb') do |f|
-        f.write(output)
-      end
+      File.write(path, output)
 
       trigger_hooks(:post_write)
     end
 
     # Whether the file is published or not, as indicated in YAML front-matter
     #
-    # Returns true if the 'published' key is specified in the YAML front-matter and not `false`.
+    # Returns true if the 'published' key is
+    # specified in the YAML front-matter and not `false`.
     def published?
-      !(data.key?('published') && data['published'] == false)
+      !(data.key?("published") && data["published"] == false)
```

Comment:
The comment, I think.

```ruby
# Returns 'false' if the 'published' key is specified in the YAML front-matter and is 'false'. Otherwise returns 'true'.
```

¯\\\_(ツ)_/¯